design/button.behavior.md

```diff
@@ -0,0 +1,70 @@
+# Button Behavior
+
+Button behavior is the high level representation of how an MRE actor should behave as a button.  This behavior consists of two actions
+that occur on the object when it is targeted and when it is activated or clicked.  As part of these interactions, the point of targeting
+of the button will be passed along as additional data to the app for consumption.
+
+Button Behavior Actions:
+- `hover` - The action that occurs when the button is targeted by the host app input system. 
+    - Hover states: `enter`, `hovering`, `exit`
+- `click` - The action that occurs when a complete click has happened on the button consisting of both a button pressed and button released event. 
+    - Click states: None.  This is a single atomic event.
+- `button` - The action that occurs when the button interaction state has changed from being pressed and then released.
+    - Button states: `pressed`, `holding`, `released`
+    
+Button Event Data:
+- `appSpaceTargetPoints` - The collection of target points for the current state event for the action.
+
+Button Event Data Per Action:
+- Hover Action:
+	- `enter` - single app coordinate target point for where the hover is first entered.
+	- `hovering` - collection of app coordinate target points for the time while hovering is occuring, sent at a set frequency.
+	- `exit` - single app coordinate target point for where the hover was exited.
+- Click Action:
+	- There is only the click state and will contain the single app coordinate target point of the click action.
+- Button Action:
+	- `pressed` - single app coordinate target point for where the pressed event began.
+	- `holding` - collection of app coordinate target points for the time while the button is held down, sent at a set frequency.
+	- `released` - single app coordinate target point for where the button is released.
+	
+## Architecture
+
+### ButtonEventData
+``` ts
+interface ButtonEventData {
+	appSpaceTargetPoints: TransformLike[];
```

Comment:
Do points ever have rotation or scale? Why is this a TransformLike and not a Vector3?

Reply:
I haven't pushed the update yet, but changed this to a vector 3 as point is the data we get and care about.